src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs

```diff
@@ -123,10 +116,51 @@ public int Run(ParseResult parseResult)
             return hasFailed ? ExitCodes.GenericFailure : ExitCodes.Success;
         }
 
-        private void WaitOnMSBuildHandlerPipeConnectionLoop()
+        private bool RunMSBuild(ParseResult parseResult, bool allowBinLog)
+        {
+            int msbuildResult;
+
+            if (parseResult.HasOption(TestingPlatformOptions.ProjectOption))
+            {
+                string filePath = parseResult.GetValue(TestingPlatformOptions.ProjectOption);
+
+                if (!File.Exists(filePath))
+                {
+                    VSTestTrace.SafeWriteTrace(() => LocalizableStrings.CmdNonExistentProjectFilePathDescription);
+                    return false;
+                }
+
+                msbuildResult = _msBuildConnectionHandler.RunWithMSBuild(filePath, allowBinLog);
+            }
+            else
+            {
+                // If no filter was provided neither the project using --project,
+                // MSBuild will get the test project paths in the current directory
+                msbuildResult = _msBuildConnectionHandler.RunWithMSBuild(allowBinLog);
+            }
+
+            if (msbuildResult != 0)
+            {
+                VSTestTrace.SafeWriteTrace(() => $"Get projects properties with msbuild  didn't execute properly with exit code: {msbuildResult}.");
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool BinLogEnabled(List<string> args)
```

Comment:
Nit: It's considered a good practice to have `Is`, `Has`, `Can` prefix for bool methods